language/move-prover/src/bytecode_translator.rs

```diff
@@ -600,7 +645,8 @@ impl<'env> ModuleTranslator<'env> {
             | FunctionEntryPoint::VerificationDefinition
             | FunctionEntryPoint::DirectIntraModule
             | FunctionEntryPoint::Indirect
-            | FunctionEntryPoint::DirectInterModule => "{:inline 1} ".to_string(),
+            | FunctionEntryPoint::DirectInterModule
+            | FunctionEntryPoint::Mutated(_) => "{:inline 1} ".to_string(),
```

Comment:
Why do we need a new entry point kind? This is strange and will lead to confusion, because those entry points have a rather specific semantics. Similar as you extended ConditionKind, this will also leak over the place and increase the complexity of the code model.